### DISCUSSION TITLE
[question] how to make the popup untouchable by the cursor

### DISCUSSION THREAD

**Discussion by alexzanderr** [MunifTanjim/nui.nvim#42] [Q&A]
```lua
local Popup = require("nui.popup")
local event = require("nui.utils.autocmd").event

-- https://github.com/MunifTanjim/nui.nvim/tree/main/lua/nui/popup
local popup = Popup({
    enter = false,
    focusable = false,
    relative = "win",
    border = {},
    position = {
        row = 37,
        col = 3,
    },
	-- zindex = 50,
    size = {
        width = 150,
        height = 1,
    },
    buf_options = {
        modifiable = true,
        readonly = false,
    },
    win_options = {
        winblend = 10,
        winhighlight = "Normal:Normal",
    },
})





function FloatingWindow()
    -- mount/open the component
    popup:mount()

    -- unmount component when cursor leaves buffer
	--     popup:on("InsertLeave", function()
	--
	-- )
	local lsp_status = "class Something > def __init__ > for i in range(3): > self.x = 3"
	vim.api.nvim_buf_set_lines(popup.bufnr, 0, 1, false, {
		lsp_status,
	})


    -- set content

end

vim.cmd("autocmd InsertLeave * lua FloatingWindow()")
```

i tried to create a floating window that displays the current tree sitter location
![image](https://user-images.githubusercontent.com/54772464/134786508-5e4626de-c7f0-4863-9b42-e2a652177435.png)

its doing great. but the problem is that i can hover with the cursor over it.
![image](https://user-images.githubusercontent.com/54772464/134786534-063b0c31-8fb2-402d-97e6-44d8f0f15b43.png)

in above photo, my cursor is actually on the 
![image](https://user-images.githubusercontent.com/54772464/134786537-615e8bec-23c1-4214-bd11-df309d619403.png)

print statement, but also the floating window its over it, its floating.

the question is, how do i make this floating window `rock solid` just like statusline (to not hover with the cursor anymore and not to see background text in it (make it none transparent))

**Comment by alexzanderr** [comment]
btw, very cool plugin and this is not an issue, this is more a documentation request

**Comment by MunifTanjim** [comment]
> not to see background text in it (make it none transparent)

That's what `winblend` does. Set `winblend = 0` (check: `:h 'winblend'`)

> how do i make this floating window rock solid just like statusline

Are you sure you want a floating window here? Maybe you're looking for something else? I'm not completely sure about exactly what behavior you're looking for... are you sure it can be done with Neovim?

If you know any plugins that does similar thing, you can take a look at their source code to see how it's done.

@alexzanderr

**Comment by alexzanderr** [comment]
> Are you sure you want a floating window here?

very sure

i've come up with `set scrolloff = 1`, this helps partially, but when going to the end of file on the last line, its overlaying

**Comment by alexzanderr** [comment]
btw another related problem with this plugin:

when using this config:
```lua
local lines = vim.fn.winheight("%")
local columns = vim.fn.winwidth("%")
local treesitter_statusline = Popup({
        enter = false,
        focusable = false,
        relative = "editor",
        border = {},
        position = {
            row = lines,
            col = 0,
        },
        -- zindex = 50,
        size = {
            width = columns,
            height = 1,
        },
        buf_options = {
            modifiable = true,
            readonly = false,
        },
		-- transparency
        -- win_options = {
        --     winblend = 0,
        --     winhighlight = "Normal:Normal",
        -- },

    })
treesitter_statusline:mount()
```

i have this problem:
![image](https://user-images.githubusercontent.com/54772464/134811398-f30abe5c-b913-491f-abee-0a7fef633a24.png)

my nvim statusline its covered

normally it looks like this
![image](https://user-images.githubusercontent.com/54772464/134811423-95dad82e-cec4-4d03-8ede-6fce7f324930.png)


but when i use `hardcoded` values (just for `row`; row has a problem with hardcoded values) for Popup setup function:
```lua
    local treesitter_statusline = Popup({
        enter = false,
        focusable = false,
        relative = "editor",
        border = {},
        position = {
            row = 38,
            col = 0,
        },
        -- zindex = 50,
        size = {
            width = columns,
            height = 1,
        },
        buf_options = {
            modifiable = true,
            readonly = false,
        },
		-- transparency
        -- win_options = {
        --     winblend = 0,
        --     winhighlight = "Normal:Normal",
        -- },

    })
    treesitter_statusline:mount()
```

![image](https://user-images.githubusercontent.com/54772464/134811487-188966a9-5e9b-448c-bc40-bac3571e62c4.png)

its very fine


notice that when i use these:
```lua
	local lines = vim.fn.winheight("%")
	local columns = vim.fn.winwidth("%")
```

instead of these:
```lua
        position = {
            row = 38,
            col = 0,
        },
```
it has a problem with overlaying.


why?

**Reply by MunifTanjim** [reply]
What's the value of `lines = vim.fn.winheight("%")`? Have you tried seeing that?

```lua
local lines = vim.fn.winheight("%")
print(lines)
```

**Reply by alexzanderr** [reply]
yes ofc i tried seeing that. these are the values for vim current window configuration. 

in my case `lines = 38` (same as hardcoded values)

**Comment by alexzanderr** [comment]
but i dont understand why it has bugs when using `	local lines = vim.fn.winheight("%")` instead of harcoding the value

**Reply by alexzanderr** [reply]
its the same thing. its an integer.

**Comment by alexzanderr** [comment]
the problem was `-- vim.cmd("autocmd BufEnter * lua TreeSitterStatusline()")`

actually, the options for popup where `vim.fn.winheight("%")` meaning that it gets the vim editor total lines because it was `relative = "win"`

so the lua file was creating a popup on buf enter event, at that time, bufferline is not loaded. so the total number of lines was `lines - 1`. the status line was covered because it was another popup was wasnt mounted.

so in total there were 2 popups.
